### ISSUE TITLE
iterating elements

### ISSUE THREAD

**Issue by filchristou** [JuliaGraphs/MetaGraphsNext.jl#32]
I find iterating through the elements of a `MetaGraph` hard.
For the docs example:
```julia
julia> colors = MetaGraph( Graph(), VertexData = String, EdgeData = Symbol, graph_data = "graph_of_colors")
Meta graph based on a {0, 0} undirected simple Int64 graph with vertex labels of type Symbol, vertex metadata of type String, edge metadata of type Symbol, graph metadata given by "graph_of_colors", and default weight 1.0
julia> colors[:red] = "warm";
julia> colors[:yellow] = "warm";
julia> colors[:blue] = "warm";  # wrong metadata
julia> colors[:blue] = "cool";
julia> colors[:red, :yellow] = :orange;
julia> colors[:red, :blue] = :violet;
julia> colors[:yellow, :blue] = :green;
```
I would do
```julia
julia> [colors[label_for(colors,v)] for v in vertices(colors)]
3-element Vector{String}:
 "warm"
 "warm"
 "cool"
```
which is a little bit tedious to write and I would love some syntactic sugar like:
```julia
[get_vertex(colors, v) for v in vertices(colors)]
```
The iteration of edges is even more tedious to write:
```julia
julia> [colors[label_for(colors, e.src), label_for(colors, e.dst)] for e in edges(colors)]
3-element Vector{Symbol}:
 :orange
 :violet
 :green
```

Ofc it's also ridiculously simple to write such functions alone, but I think such functionality should be provided from the library. 

Or is there another easier to way to iterate `MetaGraph` elements already ?

**Comment by filchristou** [comment]
Locally I create a function which returns all the node-data like following:
```julia
julia> get_vertices(x) = getindex.(values(x.vertex_properties), 2)
julia> get_vertices(colors)
3-element Vector{String}:
 "warm"
 "warm"
 "cool"
```

**Comment by bramtayl** [comment]
Well, you can use the fields of the objects directly, but that's not really a public interface. I agree, a public interface would be nice. Maybe something like

`EdgeIterator` which would collect to

`[:blue => "cool", :red => "warm", :yellow => "warm"]`

and

`VertexIterator` which would collect to

`[(:red, :yellow) => :orange, (:red, :blue) => :violet, (:yellow, :blue) => :green)]`

?

`

**Comment by filchristou** [comment]
Probably you meant it the other way around, but yes, something like that would prevent boilerplate code.

**Comment by gdalle** [comment]
> Well, you can use the fields of the objects directly, but that's not really a public interface. I agree, a public interface would be nice. Maybe something like
> 
> `EdgeIterator` which would collect to
> 
> `[:blue => "cool", :red => "warm", :yellow => "warm"]`
> 
> and
> 
> `VertexIterator` which would collect to
> 
> `[(:red, :yellow) => :orange, (:red, :blue) => :violet, (:yellow, :blue) => :green)]`
> 
> ?
> 
> `

This sounds good, gonna take a look

**Comment by gdalle** [comment]
Related: #39 and #44

**Comment by gdalle** [comment]
We should also add `edge_labels` and `(in/out)neighbor_labels`

**Comment by gdalle** [comment]
And make sure that the order is coherent with the iterators `vertices`, `edges` and `(in/out)neighbors`

**Comment by gdalle** [comment]
thanks @thchr

**Comment by gdalle** [comment]
Solved by #50
